mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -471,6 +484,59 @@ class LinearRegressionSummary private[regression] (
     predictions.select(t(col(predictionCol), col(labelCol)).as("residuals"))
   }
 
+  lazy val numInstances: Long = predictions.count()
+
+  lazy val dfe = if (model.getFitIntercept) {
```

Comment:
Keep this one private, or a more descriptive name? We need explicit types for public/private members.